src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -948,6 +953,100 @@ class spell_warr_vigilance_trigger : public SpellScriptLoader
         }
 };
 
+// Heroic leap - 6544
+class spell_warr_heroic_leap : public SpellScriptLoader
+{
+public:
+    spell_warr_heroic_leap() : SpellScriptLoader("spell_warr_heroic_leap") { }
+
+    class spell_warr_heroic_leap_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_heroic_leap_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_HEROIC_LEAP_JUMP))
+                return false;
+            return true;
+        }
+
+        SpellCastResult CheckElevation()
+        {
+            if (WorldLocation const* dest = GetExplTargetDest())
+            {
+                if (dest->GetPositionZ() > GetCaster()->GetPositionZ() + 4.0f)
+                    return SPELL_FAILED_NOPATH;
+                else if (GetCaster()->HasAuraType(SPELL_AURA_MOD_ROOT))
```

Comment:
But those should not result in SPELL_FAILED_ROOTED. SPELL_FAILED_ROOTED is only for auras with SPELL_AURA_MOD_ROOT. 

Client doesn't say "Can't be done while rooted" when having stun auras.